From c233501fd7632f5a0f354d836616c8fa99e2ae8d Mon Sep 17 00:00:00 2001
From: LibXZR <i@xzr.moe>
Date: Mon, 13 Jun 2022 19:50:38 +0800
Subject: [PATCH 2/7] NetworkTraffic: General improvements

- Remove unused resource
- Fix indent
- Use fixed width to avoid moving around
- Use INVISIBLE rather than GONE for auto hiding to reserve space
- Add missing makeVisible() for uplink
- Cleanup and optimize

Change-Id: Ic3b6021feeba6c2ee7b2a5f26fda7d7c4eea67cf
Signed-off-by: LibXZR <i@xzr.moe>
---
 .../ButtonPreference/res/values/attrs.xml     |  2 +-
 .../stat_sys_network_traffic_updown.xml       | 29 -------------------
 packages/SystemUI/res/values/arrow_dimens.xml |  3 ++
 .../statusbar/policy/NetworkTraffic.java      | 20 ++++++++-----
 4 files changed, 16 insertions(+), 38 deletions(-)
 delete mode 100644 packages/SystemUI/res/drawable/stat_sys_network_traffic_updown.xml

diff --git a/packages/SettingsLib/ButtonPreference/res/values/attrs.xml b/packages/SettingsLib/ButtonPreference/res/values/attrs.xml
index 9a4312aa8b36..944cffbb8793 100644
--- a/packages/SettingsLib/ButtonPreference/res/values/attrs.xml
+++ b/packages/SettingsLib/ButtonPreference/res/values/attrs.xml
@@ -19,4 +19,4 @@
     <declare-styleable name="ButtonPreference">
         <attr name="android:gravity" />
     </declare-styleable>
-</resources>
\ No newline at end of file
+</resources>
diff --git a/packages/SystemUI/res/drawable/stat_sys_network_traffic_updown.xml b/packages/SystemUI/res/drawable/stat_sys_network_traffic_updown.xml
deleted file mode 100644
index 4e0b63ceffe4..000000000000
--- a/packages/SystemUI/res/drawable/stat_sys_network_traffic_updown.xml
+++ /dev/null
@@ -1,29 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<!--
-    Copyright (C) 2017 The Android Open Source Project
-    Copyright (C) 2017 ABC rom
-
-    Licensed under the Apache License, Version 2.0 (the "License");
-    you may not use this file except in compliance with the License.
-    You may obtain a copy of the License at
-
-        http://www.apache.org/licenses/LICENSE-2.0
-
-    Unless required by applicable law or agreed to in writing, software
-    distributed under the License is distributed on an "AS IS" BASIS,
-    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-    See the License for the specific language governing permissions and
-    limitations under the License.
--->
-<vector xmlns:android="http://schemas.android.com/apk/res/android"
-        android:width="5dp"
-        android:height="17.0dp"
-        android:viewportWidth="7.1"
-        android:viewportHeight="24.0">
-    <path
-        android:fillColor="#ffffff"
-        android:pathData="M3.6000004,10.8l3.5,0.0 -3.5,-7.3 0.0,0.0 0.0,0.0 0.0,0.0 0.0,0.0 -3.6,7.3z"/>
-    <path
-        android:fillColor="#ffffff"
-        android:pathData="M3.6000004,13.2l-3.6,0.0 3.6,7.3 0.0,0.0 0.0,0.0 0.0,0.0 0.0,0.0 3.5,-7.3z"/>
-</vector>
diff --git a/packages/SystemUI/res/values/arrow_dimens.xml b/packages/SystemUI/res/values/arrow_dimens.xml
index 122d9bd589c0..1d11a4da786e 100644
--- a/packages/SystemUI/res/values/arrow_dimens.xml
+++ b/packages/SystemUI/res/values/arrow_dimens.xml
@@ -19,4 +19,7 @@
 
     <dimen name="volume_dialog_panel_transparent_padding_left">8dp</dimen>
 
+    <!-- Network traffic -->
+    <dimen name="network_traffic_width">25sp</dimen>
+    
 </resources>
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/policy/NetworkTraffic.java b/packages/SystemUI/src/com/android/systemui/statusbar/policy/NetworkTraffic.java
index 458bc6c75232..6f4b474813db 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/policy/NetworkTraffic.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/policy/NetworkTraffic.java
@@ -47,6 +47,8 @@ public class NetworkTraffic extends TextView {
     private static final int GB = MB * KB;
     private static final String symbol = "/s";
 
+    private final int mWidth;
+
     protected boolean mIsEnabled;
     private boolean mAttached;
     private long totalRxBytes;
@@ -84,7 +86,7 @@ public class NetworkTraffic extends TextView {
 
             if (shouldHide(rxData, txData, timeDelta)) {
                 setText("");
-                setVisibility(View.GONE);
+                setVisibility(View.INVISIBLE);
                 mVisible = false;
             } else if (shouldShowUpload(rxData, txData, timeDelta)) {
                 // Show information for uplink if it's called for
@@ -94,6 +96,7 @@ public class NetworkTraffic extends TextView {
                 if (output != getText()) {
                     setText(output);
                 }
+                makeVisible();
             } else {
                 // Add information for downlink if it's called for
                 CharSequence output = formatOutput(timeDelta, rxData, symbol);
@@ -125,7 +128,7 @@ public class NetworkTraffic extends TextView {
             SpannableString spanUnitString;
             SpannableString spanSpeedString;
 
-            if (speed >= GB) {
+            if (speed >= 1000 * MB) {
                 unit = "GB";
                 decimalFormat = new DecimalFormat("0.00");
                 formatSpeed =  decimalFormat.format(speed / (float)GB);
@@ -137,7 +140,7 @@ public class NetworkTraffic extends TextView {
                 decimalFormat = new DecimalFormat("00.0");
                 unit = "MB";
                 formatSpeed =  decimalFormat.format(speed / (float)MB);
-            } else if (speed >= MB) {
+            } else if (speed >= 1000 * KB) {
                 decimalFormat = new DecimalFormat("0.00");
                 unit = "MB";
                 formatSpeed =  decimalFormat.format(speed / (float)MB);
@@ -174,7 +177,7 @@ public class NetworkTraffic extends TextView {
 
         private boolean shouldShowUpload(long rxData, long txData, long timeDelta) {
             long speedRxKB = (long)(rxData / (timeDelta / 1000f)) / KB;
-                long speedTxKB = (long)(txData / (timeDelta / 1000f)) / KB;
+            long speedTxKB = (long)(txData / (timeDelta / 1000f)) / KB;
 
             return (speedTxKB > speedRxKB);
         }
@@ -210,6 +213,7 @@ public class NetworkTraffic extends TextView {
         super(context, attrs, defStyle);
         final Resources resources = getResources();
         mTintColor = getCurrentTextColor();
+        mWidth = resources.getDimensionPixelSize(R.dimen.network_traffic_width);
         setMode();
         Handler mHandler = new Handler();
         mConnectivityManager =
@@ -311,10 +315,11 @@ public class NetworkTraffic extends TextView {
                 lastUpdateTime = SystemClock.elapsedRealtime();
                 mTrafficHandler.sendEmptyMessage(1);
             }
+            if (mAutoHideThreshold == 0)
+                makeVisible();
             return;
-        } else {
-            clearHandlerCallbacks();
         }
+        clearHandlerCallbacks();
         setVisibility(View.GONE);
         mVisible = false;
     }
@@ -331,8 +336,7 @@ public class NetworkTraffic extends TextView {
         setMaxLines(2);
         setSpacingAndFonts();
         updateTrafficDrawable();
-        setVisibility(View.GONE);
-        mVisible = false;
+        setWidth(mWidth);
     }
 
     private void clearHandlerCallbacks() {
-- 
2.32.0

